src/app/clusters/fabric/fabric.cpp

```diff
@@ -32,6 +35,108 @@
 #include "gen/command-id.h"
 
 using namespace chip;
+using namespace ::chip::DeviceManager;
+using namespace ::chip::DeviceLayer;
+
+CHIP_ERROR addNewAdmin(AdminPa)
+
+    EmberAfStatus writeFabricAttribute(uint8_t endpoint, AttributeId attributeId, uint8_t * buffer, int32_t index = -1)
+{
+    EmberAfAttributeSearchRecord record;
+    record.endpoint         = endpoint;
+    record.clusterId        = ZCL_FABRIC_CLUSTER_ID;
+    record.clusterMask      = CLUSTER_MASK_SERVER;
+    record.manufacturerCode = EMBER_AF_NULL_MANUFACTURER_CODE;
+    record.attributeId      = attributeId;
+
+    // When reading or writing a List attribute the 'index' value could have 3 types of values:
+    //  -1: Read/Write the whole list content, including the number of elements in the list
+    //   0: Read/Write the number of elements in the list, represented as a uint16_t
+    //   n: Read/Write the nth element of the list
+    //
+    // Since the first 2 bytes of the attribute are used to store the number of elements, elements indexing starts
+    // at 1. In order to hide this to the rest of the code of this file, the element index is incremented by 1 here.
+    // This also allows calling writeAttribute() with no index arg to mean "write the length".
+    emberAfDoorLockClusterPrintln("Fabric: Writing element");
+    return emAfReadOrWriteAttribute(&record, NULL, buffer, 0, true, index + 1);
+}
+
+EmberAfStatus writeFabric(FabricId fabricId, NodeId nodeId, uint16_t vendorId, int32_t index)
+{
+    emberAfDoorLockClusterPrintln("Fabric: Trying to write a new fabric");
+    EmberAfStatus status    = EMBER_ZCL_STATUS_SUCCESS;
+    AttributeId attributeId = ZCL_FABRICS_LIST_ATTRIBUTE_ID;
+
+    uint16_t fabricCount = 1;
+    EmberAfFabricDescriptor fabricDescriptor;
+    fabricDescriptor.FabricId = fabricId;
+    fabricDescriptor.NodeId   = nodeId;
+    fabricDescriptor.VendorId = vendorId;
+
+    status = writeFabricAttribute(1, attributeId, (uint8_t *) &fabricDescriptor, index);
+    VerifyOrReturnError(status == EMBER_ZCL_STATUS_SUCCESS, status);
+    return writeFabricAttribute(1, attributeId, (uint8_t *) &fabricCount);
```

Comment:
Why `1` ? I would expect the fabric cluster to be on endpoint `0`. You probably want to pass the `endpoint` as a parameter.